src/objectprotocol/number.rs

```diff
@@ -0,0 +1,323 @@
+    /// throw an exception on any version before that.
+    ///
+    /// See [PEP 0456](https://www.python.org/dev/peps/pep-0465/) for details.
+    #[inline]
+    fn matrix_multiply(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        #[cfg(not(any(feature = "python3-4", feature = "python2-sys")))] {
+            other.with_borrowed_ptr(py, |other| unsafe {
+                err::result_from_owned_ptr(py, ffi::PyNumber_MatrixMultiply(self.as_ptr(), other))
+            })
+        }
+        #[cfg(any(feature = "python3-4", feature = "python2-sys"))] {
+
+            drop(other);
+            Err(crate::PyErr::new::<crate::exc::TypeError, _>(
+                py,
+                "Matrix multiplication is unsupported before Python 3.5"
+            ))
+        }
+    }
+    /// Perform exponentiation, equivalent to the Python expression `self ** other`,
+    /// or the two-argument form of the builtin method pow: `pow(self, other)`
+    ///
+    /// Invokes the `__pow__` magic-method
+    ///
+    /// See also [NumberProtocol::pow_mod].
+    #[inline]
+    fn pow(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        self.pow_mod(py, other, py.None())
+    }   
+    /// Perform exponentiation modulo an integer,
+    /// mathematically equivalent to `self ** other % mod`
+    /// but computed much more efficiently.
+    /// 
+    /// Equivalent to invoking the three-argument form
+    /// of the builtin `pow` method: `pow(self, other, z)`
+    ///
+    /// Invoking with a `None` for modulo is equivalent to
+    /// the regular power operation.
+    ///
+    /// Invokes the `__pow__` magic-method
+    #[inline]
+    fn pow_mod(&self, py: Python, exp: impl ToPyObject, z: impl ToPyObject) -> PyResult<PyObject> {       
+        exp.with_borrowed_ptr(py, |exp| {
+            z.with_borrowed_ptr(py, |z| unsafe {
+                err::result_from_owned_ptr(py, ffi::PyNumber_Power(self.as_ptr(), exp, z))
+            })
+        })
+    }
+    /// Perform the "true division" operation,
+    /// equivalent to the Python expression `self / other`,
+    /// 
+    /// Invokes the `__truediv__` magic-method.
+    #[inline]   
+    fn true_div(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        other.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_TrueDivide(self.as_ptr(), other))
+        })
+    }
+    /// Perform the "floor division" operation,
+    /// equivalent to the Python expression `self // other`,
+    /// 
+    /// This method was added in Python 3.
+    /// If compiling against Python 2, it unconditional throws an error.
+    ///
+    /// Invokes the `__floordiv__` magic-method.
+    #[inline]
+    fn div_floor(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {       
+        other.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_TrueDivide(self.as_ptr(), other))
```

Comment:
Should this be calling `PyNumber_FloorDivide`?

Reply:
Fixed. Thank you!